nltk/metrics/distance.py

```diff
@@ -366,10 +366,74 @@ def jaro_winkler_similarity(s1, s2, p=0.1, max_l=4):
     return jaro_sim + (l * p * (1 - jaro_sim))
 
 
+def hamming_distance(s1, s2):
+    """
+    The Hamming distance measures the distance between two equal length strings
+    and it is determined by counting the positions where the two strings have different
+    characters.
+
+    Source(s):
+
+        (1) https://en.wikipedia.org/wiki/Hamming_distance
+        (2) Hamming, R. W. (April 1950). "Error detecting and error correcting codes".
+            The Bell System Technical Journal. 29 (2): 147–160.
+
+    :param s1, s2: The strings to be analysed
+    :type s1: str
+    :type s2: str
+    :rtype int
+
+    """
+    assert len(s1) == len(s2), "Strings must be the same length."
+    return sum(i != j for i, j in zip(s1, s2))
+
+
+def lee_distance(s1, s2):
+    """
+    Lee Distance metric calculates the distance between two strings by re-encoding them
+    using a q-ary alphabet of all the unique characters present in one or both strings.
+    Then using the q-ary alphabet index values provides the sum of
+
+        min(| string1_i - string2_i | , alphabet_size - | string1_i - string2_i |)
+
+    for each value i in the length of string1 or string2 (these are interchangeable because
+    both strings must be of equivalent length).
+
+    Source(s):
+
+        (1) https://en.wikipedia.org/wiki/Lee_distance
+
+    :param s1, s2: The strings to be analysed
+    :type s1: str
+    :type s2: str
+    :rtype int
+
+    """
+    assert len(s1) == len(s2), "Strings must be the same length."
+
+    alphabet = sorted(set(s1).union(set(s2)))
+    q = len(alphabet)
+
+    assert q > 1, "number of distinct characters must be greater than 1"
+
+    s1_enc = _lee_string_encoder(s1, alphabet=list(alphabet))
+    s2_enc = _lee_string_encoder(s2, alphabet=list(alphabet))
```

Comment:
Casting to list is not necessary here, `sorted()` returns a list. https://docs.python.org/3/library/functions.html#sorted